### DISCUSSION TITLE
I want the dot to change direction when it hits the wall. I don't know what the problem is.

### DISCUSSION THREAD

**Discussion by jengelbertt** [ManimCommunity/manim#3846] [Q&A] [answered]
class FunctionTracpo(Scene):
    def construct(self):

        width=5
        height=5
        box=Rectangle(height=height , width=width )
        box.set_stroke(width=2)

        ball=Dot()
        ball.vx , ball.vy = 0.05 , 0.05

        def update_circle(c):
            right_point=c.get_right()[0]
            left_point=c.get_left()[0]
            top_point=c.get_top()[1]
            bottom_point=c.get_bottom()[1]

            if right_point >= width/2 or left_point <= -width/2:
                c.vx *=-1
            
            if top_point > height/2 or bottom_point < -height/2:
                c.vy *=-1

            c.shift(c.vx*RIGHT + c.vy*UP)

        self.play(
            Create(ball), Create(box) ,
            UpdateFromFunc(ball,update_circle)
        )

        self.wait(5)

**Comment by uwezi** [comment]
can you please format your code correctly by enclosing it between triple backticks as described here:
![image](https://github.com/ManimCommunity/manim/assets/8582807/5865a650-333d-4358-b423-78f87657075e)

**Comment by uwezi** [comment, marked as answer]
ok, here is a working code:

```py
class FunctionTracpo(Scene):
    def construct(self):

        width=5
        height=5
        box=Rectangle(height=height , width=width )
        box.set_stroke(width=2)

        ball=Dot()
        ball.vx , ball.vy = 0.05 , 0.05

        def update_circle(c):
            right_point=c.get_right()[0]
            left_point=c.get_left()[0]
            top_point=c.get_top()[1]
            bottom_point=c.get_bottom()[1]

            if right_point >= width/2 or left_point <= -width/2:
                c.vx *=-1

            if top_point > height/2 or bottom_point < -height/2:
                c.vy *=-1


            c.shift(c.vx*RIGHT + c.vy*UP)

        self.play(
            Create(ball), Create(box) ,
        )
        ball.add_updater(update_circle)
        self.wait(5, frozen_frame=False)
```

**Reply by jengelbertt** [reply]
Thank you very much man .

**Comment by uwezi** [comment]
You might want to modify your direction-changing equations a bit... we recently had a similar discussion on Discord...
In the current form your dot might get trapped under some conditions...

**Reply by jengelbertt** [reply]
I am also eager to join you in Discord. Where should I come?

**Reply by uwezi** [reply]
> I am also eager to join you in Discord. Where should I come?

just follow the link: https://docs.manim.community/en/stable/faq/general.html?highlight=discord#where-can-i-find-more-resources-for-learning-manim
